src/main/java/cubicchunks/network/PacketCubeBlockChange.java

```diff
@@ -68,14 +70,14 @@ public PacketCubeBlockChange(Cube cube, TShortCollection localAddresses) {
             int y = AddressTools.getLocalY(localAddress);
             int z = AddressTools.getLocalZ(localAddress);
             this.blockStates[i] = cube.getBlockState(x, y, z);
-            xzAddresses.add(x | z << 4);
+            xzAddresses.add(x | z << 4); // TODO: Is this the right way around?
```

Comment:
This value ```x | z << 4``` is used in height map, therefore ```z``` value should be shifted to yours Cube.BIT_SIZE (5 for size = 32) or bits from "x" and "z" will overlap with each others.

Reply:
In a 16³ cube, both x and z are at most 4 bits in length. Thus the order in which they are packed doesn't matter in that regard or am I missing something?

Reply:
They are unpacket in a same <init> method at a line 80. Order matters here. Also note that height itself stuffed inside that value as well. So you would need to change it to ```v |= height << Cube.BIT_SIZE*2;```

Reply:
@Cyclonit also look how Client network handler handle ```int[] heightValues;``` field.

Reply:
This should probably be changed to use a util method in class like AddressTools, together with all other similar things.

Reply:
Wouldn't this just be AddressTools.getLocalAddress(int localX, int localZ)?

Reply:
Probably, I haven't looked at AddressTools in a while.